src/Login/actions.js

```diff
@@ -39,9 +39,9 @@ export function attemptLogin(uid, password) {
             credentials: 'same-origin',
             body: JSON.stringify({uid: uid, password: password})
         });
-        console.log('res: ', res);
-        const auth = await query({query: '{ auth }'});
-        console.log('auth test: ', auth);
+        const data = await query({query: '{ auth }'});
+        if(data.auth)
+            window.location = '/main';
```

Comment:
If you directly manipulate the window.location we will lose the browser history. Instead, you could use React router's `browserHistory`

`browserHistory.push('/main')`;